src/main/java/com/algorithmia/client/HttpClient.java

```diff
@@ -193,4 +234,24 @@ private HttpResponse execute(HttpUriRequest request) throws APIException {
         return client.execute(new BasicAsyncRequestProducer(target, request), consumer, null);
     }
 
+    /**
+     *   A consumer that drops the body of a response. It's useful when you just want the HTTP headers.
+     */
+    static class HttpResponseConsumer extends AsyncByteConsumer<HttpResponse> {
```

Comment:
could use a clearer name, something like HttpResponseHeaderConsumer would better indicate what this does